### PR TITLE
Add Amazon MSK guide

### PR DESCRIPTION
**Pull request by bernscode**
Add Amazon MSK guide.

## Info
- **Description**: 
Add Amazon MSK guide.

- **Preview**: 
https://pr-903.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/connector-amazon-msk/

- **Related code PR**: 
N/A

- **Related doc issue**: 
Resolves no doc issue.

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/guides/connector-amazon-msk.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+### Create a source in RisingWave
+
+To learn about the specific syntax used to consume data from a Kafka topic, see [Ingest data from Kafka](https://pr-878.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/create-source-kafka/).
+
+For example, the following query creates a materialized source that consumes data from an MSK topic connected to Kafka.
+
+
+```sql
+create materialized source s (v1 int, v2 varchar) with (
```
Comment (emile-00) [inline docs/guides/connector-amazon-msk.md]: Remember to use the conventional capitalization format for this SQL query.

Comment (bernscode) [reply docs/guides/connector-amazon-msk.md]: @emile-00 Good catch! I've made the update on the latest commit. Thanks!

**Review by emile-00 — APPROVED** [review]

File: docs/guides/connector-amazon-msk.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+title: Ingest data from Amazon MSK
+description: Ingest data from Amazon MSK.
+slug: /connector-amazon-msk
+---
+
+Amazon Managed Streaming for Apache Kafka (MSK) is a fully managed service that simplifies the setup, scaling, and management of Apache Kafka clusters, a popular open-source distributed streaming platform. Kafka is designed to handle real-time data feeds and follows the publisher-subscriber (pub-sub) model. Kafka's ability to handle high-volume real-time data makes it crucial for data pipelines, analytics, and event-driven architectures.
+
+To ingest data from Amazon MSK into RisingWave, you need an operational Amazon MSK cluster and a Kafka topic established. Once set, you'll leverage the Kafka connector in RisingWave to consume data from your MSK topic.
+
+This guide will detail the ingesting streaming data from Amazon MSK into RisingWave, enabling powerful and real-time data analytics in your workflows.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: powerful and real-time data analytics sounds like marketing or sales language. Let's stick to point and focus on the purpose and procedures.

File: docs/guides/connector-amazon-msk.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+
+
+## Set up Amazon MSK
+
+To learn about how to set up an Amazon MSK account and create a cluster, see [Getting started using Amazon MSK](https://docs.aws.amazon.com/msk/latest/developerguide/getting-started.html). For this demo, we will assume the selection of **Quick create** for the **Cluster creation method** and **Provisioned** for the **Cluster type**. The cluster creation can take about 15 minutes.
+
+Once you have created a cluster, note down the following information regarding the cluster you want to connect to.
+
+1. Get the **VPC** value from the **All cluster settings** before clicking **Create cluster**.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: I'm confused. The paragraph above step 1 says once you have created a cluster, and then we ask people note down something "before" clicking Create cluster. Is this possible?

File: docs/guides/connector-amazon-msk.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+2. Get the **Security groups associated with VPC** from the **All cluster settings** before clicking **Create cluster**.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Same here. We probably want to separate what is needed before clicking Create cluster and after.

File: docs/guides/connector-amazon-msk.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+3. Get the **ARN** value from the **Cluster summary** after clicking on **Create cluster**.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Clicking not clicking on.

File: docs/guides/connector-amazon-msk.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+## Configure MSK Kafka
+
+
+### Enable SASL
+
+1. Access the [Amazon MSK console](https://console.aws.amazon.com/msk/) and select the MSK cluster.
+
+2. Click on the **Properties** tab, and click **Edit** in the **Security settings** section.
+
+3. Click the **SASL/SCRAM authentication** checkbox and click **Save changes**.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: click ... checkbox -> Select ...

File: docs/guides/connector-amazon-msk.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+
+### Create a topic using the broker address with SASL
+
+
+1. Access the [Amazon MSK console](https://console.aws.amazon.com/msk/) and select the cluster.
+
+2. Click **View client information** and copy the URL under **Private endpoint** for **SASL/SCRAM**. This will be your `<broker-url>` from now on.
+
+3. Run the following command using the `<broker-url>`.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: The purpose of this step is to create a topic. I think we need to call it out. "using the <broker-url> is self-explanatory and therefore can be left out.

File: docs/guides/connector-amazon-msk.md [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+    bin/kafka-topics.sh --bootstrap-server <broker-url> --list --command-config ~/client_sasl.properties
+    ```
+
+5. Insert test data.
+
+    ```terminal
+    bin/kafka-console-producer.sh --bootstrap-server <broker-url> --topic <topic-name> --producer.config ~/client_sasl.properties
+    ```
+
+6. Open another console and consume the message.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Is this step necessary?

File: docs/guides/connector-amazon-msk.md [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+
+## Consume data from Amazon MSK in RisingWave
+
+
+### Install and launch RisingWave
+
+See the [Get started](https://pr-878.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/get-started/) guide for options on how you can run RisingWave.
+
+
+### Download Kafka workload from GitHub and write to Kafka
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: this step is unnecessary for external users. For internal users testing the product, we may need this. For external users, the step for producing messages is done in MSK. We are mixing things up here.

**Review by hengm3467 — CHANGES_REQUESTED** [review]
@bernscode Please see my comments and update the draft accordingly.

File: docs/guides/connector-amazon-msk.md [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+
+### Create a symmetric key
+
+1. Access the [AWS Key Management Service (AWS KMS) console](https://console.aws.amazon.com/kms).
+
+2. Click **Create Key**, select **Symmetric**, and click **Next**.
+
+3. Give the key an **Alias** and click **Next**.
+
+4. Under **Administrative permissions**, select the **AWSServiceRoleForKafka** checkbox and click **Next**.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Just "Select **AWS...**..." would be good. Please do a global search in this topic so that it's consistent.

**Comment by bernscode** [comment]
@hengm3467 I've made the updates that we discussed including the ones recently noted. Thanks!

File: docs/guides/connector-amazon-msk.md [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+wget https://archive.apache.org/dist/kafka/2.6.2/kafka_2.12-2.6.2.tgz
+tar -xzf kafka_2.12-2.6.2.tgz
+```
+
+
+
+### Configure AWS IAM credentials on EC2
+
+
+1. Run the following command.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Always mention the purpose in an action that you want users to take. In this case, the purpose is to configure AWS credentials and default settings.

File: docs/guides/connector-amazon-msk.md [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+
+    ```
+    KafkaClient {
+    org.apache.kafka.common.security.scram.ScramLoginModule required
+      username="<your-username>"
+      password="<your-password>";
+    };
+    ```
+
+3. Run the following command.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Again, please find out the purpose of this action.

File: docs/guides/connector-amazon-msk.md [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+
+
+### Create a topic using the broker address with SASL
+
+
+1. Access the [Amazon MSK console](https://console.aws.amazon.com/msk/) and select the cluster.
+
+2. Click **View client information** and copy the URL under **Private endpoint** for **SASL/SCRAM**. This will be your `<broker-url>` from now on.
+
+3. Run the following command.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Same here. The purpose is to create a topic, and step 5 is to insert data into the topic. Therefore, step 5 can be right after this step. And step 4 is totally optional.

File: docs/guides/connector-amazon-msk.md [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+    bin/kafka-topics.sh --bootstrap-server <broker-url> --command-config ~/client_sasl.properties --create --topic <topic-name>
+    ```
+
+4. List the topics.
+
+    ```terminal
+    bin/kafka-topics.sh --bootstrap-server <broker-url> --list --command-config ~/client_sasl.properties
+    ```
+
+5. Insert test data.
```
Comment (hengm3467) [inline docs/guides/connector-amazon-msk.md]: Info I found: 

"Once you run the kafka-console-producer command, you will be prompted to enter messages into the console. Each message should be entered on a new line, and you can enter as many messages as you like.

When you are finished entering messages, you can simply close the console window or press Ctrl + C to exit the producer."

**Comment by bernscode** [comment]
@hengm3467 I've made the suggested changes to the latest commit. Thanks!

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**